File: Homework8/Hw8/Calculator/Calculator.cs
namespace Hw8.Calculator
{
    public class Calculator : ICalculator
    {
        public double Calculate(double value1, Operation operation, double value2)
            => operation switch
            {
                Operation.Plus => Plus(value1, value2),
                Operation.Minus => Minus(value1, value2),
                Operation.Multiply => Multiply(value1, value2),
                Operation.Divide => Divide(value1, value2),
                _ => throw new InvalidOperationException(Messages.InvalidOperationMessage)
            };

        public double Plus(double val1, double val2) => val1 + val2;

        public double Minus(double val1, double val2) => val1 - val2;

        public double Multiply(double val1, double val2) => val1 * val2;

        public double Divide(double val1, double val2) => val2 < double.Epsilon
            ? throw new InvalidOperationException(Messages.DivisionByZeroMessage)
            : val1 / val2;
    }
}

File: Homework6/Hw6/Calculator/Calculator.fs
module Hw6.Calculator

type CalculatorOperation =
     | Plus = 0
     | Minus = 1
     | Multiply = 2
     | Divide = 3

[<Literal>] 
let Plus = "+"

[<Literal>] 
let Minus = "-"

[<Literal>] 
let Multiply = "*"

[<Literal>] 
let Divide = "/"

[<System.Diagnostics.CodeAnalysis.ExcludeFromCodeCoverage>]
let inline calculate (value1: decimal, operation: string, value2: decimal) =
    match operation with
    | Plus -> Ok $"{value1 + value2}"
    | Minus -> Ok $"{value1 - value2}"
    | Multiply -> Ok $"{value1 * value2}"
    | Divide -> match value2 with
        | 0m -> Ok "DivideByZero"
        | _ -> Ok $"{value1 / value2}"
    | _ -> Error $"Could not parse value '{operation}'"